scripts/submit-analytics.js

```diff
@@ -0,0 +1,201 @@
+    framework = "electron";
+    frameworkVersion = context.dependencies["electron"];
+  }
+
+  const payloads = {
+    webHook: {
+      event: "install",
+      properties: {
+        token: "aab85907a13e1ff44a95be539d9942a9",
+        "JS Analytics Version": 2,
+        distinct_id: identifier,
+        "Anonymized Machine Identifier": identifier,
+        "Anonymized Application ID": sha256(__dirname),
+        Binding: "javascript",
+        Version: context.version,
+        Language: "javascript",
+        Framework: framework,
+        "Framework Version": frameworkVersion,
+        "JavaScript Engine": jsEngine,
+        "Host OS Type": os.platform(),
+        "Host OS Version": os.release(),
+        "Node.js version": process.version,
+      },
+    },
+  };
+
+  return payloads;
+}
+
+function isAnalyticsDisabled() {
+  let isDisabled = false;
+
+  // NODE_ENV is commonly used by JavaScript framework
+  if ("NODE_ENV" in process.env) {
+    isDisabled |= process.env["NODE_ENV"] === "production" || process.env["NODE_ENV"] === "test";
+  }
+
+  // If the user has specifically opted-out or if we're running in a CI environment
+  isDisabled |= "REALM_DISABLE_ANALYTICS" in process.env || "CI" in process.env;
+
+  return isDisabled;
+}
+
+/**
+ * Send collected analytics data to Realm's servers over HTTPS
+ * @param  {Object} payload analytics info
+ */
+async function dispatchAnalytics(payload) {
+  const https = require("https");
+
+  return new Promise((resolve, reject) => {
+    const webhookRequest = new Webhook().buildRequest(payload);
+
+    https
+      .get(webhookRequest, (res) => {
+        resolve({
+          statusCode: res.statusCode,
+          statusMessage: res.statusMessage,
+        });
+      })
+      .on("error", (error) => {
+        const message = error && error.message ? error.message : error;
+        const err = new Error(`Failed to dispatch analytics: ${message}`);
+        reject(err);
+      });
+  });
+}
+
+async function submitAnalytics(dryRun) {
+  if (isAnalyticsDisabled()) {
+    doLog("Analytics is disabled");
+    return;
+  }
+
+  const context = require("../../../package.json");
+  const payload = await fetchPlatformData(context);
+  doLog(`payload: ${JSON.stringify(payload)}`);
+
+  if (dryRun) {
+    doLog("Dry run; will not submit analytics");
+    return;
+  }
+
+  await Promise.all([
+    // send in analytics in the newer S3 format
+    dispatchAnalytics(payload),
+  ]);
+}
+
+const optionDefinitions = [
+  { name: "dryRun", type: Boolean, multiple: false, description: "If true, don't submit analytics" },
+  { name: "log", type: Boolean, multiple: false, description: "If true, print log messages" },
+];
+
+const options = commandLineArgs(optionDefinitions, { camelCase: true });
+
+let dryRun = false;
+if (options.dryRun) {
+  dryRun = true;
+}
+
+if (options.log) {
+  doLog = (msg) => console.log(msg);
+} else {
+  // eslint-disable-next-line no-unused-vars
+  doLog = (_) => {
```

Comment:
Or could this just be

```suggestion
  doLog = () => {
```